cuda/solver/lower_trs_kernels.cu

```diff
@@ -65,8 +152,60 @@ GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
 template <typename ValueType, typename IndexType>
 void solve(std::shared_ptr<const CudaExecutor> exec,
            const matrix::Csr<ValueType, IndexType> *matrix,
-           const matrix::Dense<ValueType> *b,
-           matrix::Dense<ValueType> *x) GKO_NOT_IMPLEMENTED;
+           gko::solver::SolveStruct *solve_struct,
+           matrix::Dense<ValueType> *trans_b, matrix::Dense<ValueType> *trans_x,
+           const matrix::Dense<ValueType> *b, matrix::Dense<ValueType> *x)
+{
+    using vec = matrix::Dense<ValueType>;
+    if (cusparse::is_supported<ValueType, IndexType>::value) {
+        ValueType one = 1.0;
+        auto handle = exec->get_cusparse_handle();
+#if (defined(CUDA_VERSION) && (CUDA_VERSION >= 9020))
+        x->copy_from(gko::lend(b));
+        GKO_ASSERT_NO_CUSPARSE_ERRORS(
+            cusparseSetPointerMode(handle, CUSPARSE_POINTER_MODE_HOST));
+        cusparse::csrsm2_solve(
+            handle, solve_struct->algorithm, CUSPARSE_OPERATION_NON_TRANSPOSE,
+            CUSPARSE_OPERATION_TRANSPOSE, matrix->get_size()[0],
+            b->get_stride(), matrix->get_num_stored_elements(), &one,
+            solve_struct->factor_descr, matrix->get_const_values(),
+            matrix->get_const_row_ptrs(), matrix->get_const_col_idxs(),
+            x->get_values(), b->get_stride(), solve_struct->solve_info,
+            solve_struct->policy, solve_struct->factor_work_vec);
+        GKO_ASSERT_NO_CUSPARSE_ERRORS(
+            cusparseSetPointerMode(handle, CUSPARSE_POINTER_MODE_DEVICE));
+#elif (defined(CUDA_VERSION) && (CUDA_VERSION < 9020))
+        GKO_ASSERT_NO_CUSPARSE_ERRORS(
+            cusparseSetPointerMode(handle, CUSPARSE_POINTER_MODE_HOST));
+        if (b->get_stride() == 1) {
+            auto temp_b = const_cast<ValueType *>(b->get_const_values());
+            cusparse::csrsm_solve(
+                handle, CUSPARSE_OPERATION_NON_TRANSPOSE, matrix->get_size()[0],
+                b->get_stride(), &one, solve_struct->factor_descr,
+                matrix->get_const_values(), matrix->get_const_row_ptrs(),
+                matrix->get_const_col_idxs(), solve_struct->solve_info, temp_b,
+                b->get_size()[0], x->get_values(), x->get_size()[0]);
```

Comment:
So apparently, it works nonetheless. the cusparse analysis function does not seem to use the num_rhs parameter. I will leave that as a parameter because as Terry suggests in the future it is possible that they use this to do this to do something more sophisticated.

So, in short. you can use not set the parameter(default would be 1) and it pass in your multiple rhs vector and it would solve without any problems.

Regarding the fact that it might cause some confusion, I will add an explanation to the parameter doc.

Reply:
Thanks for testing!
Interesting that it works without any modification. And yes, in that case, documenting this case would be good.